cpp/src/common/nvtx.hpp

```diff
@@ -42,4 +42,81 @@ void PUSH_RANGE(const char* name);
 /** Pop the latest range */
 void POP_RANGE();
 
+/** Push a named nvtx range that would be popped at the end of the object lifetime. */
+class AUTO_RANGE {
```

Comment:
I think this can be used more generally both inside of RAFT primitives and across different projects. Is there any particular reason we shouldn't push this whole file to RAFT?

Reply:
(I asked the same question in the gtests for your current RAFT PR but I haven't finished the review yet so I figured I'd ask here in comment).

Reply:
That would be fantastic! I'd definitely use it in Triton backends if it were available in RAFT.

Reply:
With your permission, I'd love to move to raft :) My PR https://github.com/rapidsai/raft/pull/401 would benefit from that indeed.

Reply:
Absolutely, I think that's a great idea!